vite-project/src/components/mealsList.jsx

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect } from "react";
+import ListItemWithBorder from "./listItemWithBorder";
+import apiurl from "../apiurl";
+function MealsList() {
+  const [meals, setMeals] = useState([]);
+  const [loading, setLoading] = useState(true);
+  useEffect(() => {
+    fetch(apiurl("/api/meals"))
+      .then((response) => response.json())
+      .then((data) => {
+        setMeals(data.data);
+        setLoading(false);
+      })
+      .catch((error) => console.error("Error fetching meals:", error));
```

Comment:
if there is an error you get stuck with the loading state, you can handle this scenario by using `finally` method